File: src/SignalR/common/SignalR.Common/src/Protocol/CancelInvocationMessage.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

using System.Collections.Generic;

namespace Microsoft.AspNetCore.SignalR.Protocol;

/// <summary>
/// The <see cref="CancelInvocationMessage"/> represents a cancellation of a streaming method.
/// </summary>
public class CancelInvocationMessage : HubInvocationMessage
{
    /// <summary>
    /// Initializes a new instance of the <see cref="CancelInvocationMessage"/> class.
    /// </summary>
    /// <param name="invocationId">The ID of the hub method invocation being canceled.</param>
    public CancelInvocationMessage(string invocationId) : base(invocationId)
    {
    }
}

File: src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/CancelInvocationMessage.java
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

package com.microsoft.signalr;

import java.util.Map;

public final class CancelInvocationMessage extends HubMessage {
    private final int type = HubMessageType.CANCEL_INVOCATION.value;
    private Map<String, String> headers;
    private final String invocationId;

    public CancelInvocationMessage(Map<String, String> headers, String invocationId) {
        if (headers != null && !headers.isEmpty()) {
            this.headers = headers;
        }
        this.invocationId = invocationId;
    }

    public Map<String, String> getHeaders() {
        return headers;
    }

    public String getInvocationId() {
        return invocationId;
    }

    @Override
    public HubMessageType getMessageType() {
        return HubMessageType.CANCEL_INVOCATION;
    }
}
